Neos.Flow/Classes/Security/Policy/Role.php

```diff
@@ -84,45 +66,48 @@ class Role
      */
     public function __construct(string $identifier, array $parentRoles = [], string $label = '', string $description = '')
     {
-        if (preg_match(self::ROLE_IDENTIFIER_PATTERN, $identifier, $matches) !== 1) {
-            throw new \InvalidArgumentException('The role identifier must follow the pattern "Vendor.Package:RoleName", but "' . $identifier . '" was given. Please check the code or policy configuration creating or defining this role.', 1365446549);
-        }
-        $this->identifier = $identifier;
-        $this->packageKey = $matches[1];
-        $this->name = $matches[2];
-        $this->label = $label ?: $matches[2];
+        $this->id = RoleId::fromString($identifier);
+        $this->label = $label ?: $this->id->getName();
         $this->description = $description;
         $this->parentRoles = $parentRoles;
     }
 
```

Comment:
```suggestion

public static function fromRoleId(RoleId $id, array $parentRoles = [], string $label = '', string $description = ''): self
{
    return new ($id->value, array $parentRoles = [], string $label = '', string $description = '');
}

public static function fromIdentifierString(string $identifier, array $parentRoles = [], string $label = '', string $description = ''): self
{
    return new ($identifier, array $parentRoles = [], string $label = '', string $description = '');
}
```

And then deprecate the __construct as it is so we can at least  open a window for refactoring internally?

Reply:
mh, `Role::fromRoleId()` and `Role::fromIdentifierString()` sounds wrong to me.. But what do you think of
```php
public function __construct(RoleId|string $id, array $parentRoles = [], string $label = '', string $description = '') {
  if (is_string($id)) {
    $id = RoleId::fromString($id);
  }
  // ...
}
```

Reply:
Sure, sounds good! My thought train was not finished really. I was thinking / hoping how we can refactor towards something like RoleIds $parentRoles or somthing, but the static constructors would not necessarily make that easier :D

Reply:
Adjusted with 32bbd2dbeeeda3bd5ea7a8daa6205d1298f01580

Reply:
Yeah, totally fair